idea/frontend/src/features/metadata/api/requests.ts

```diff
@@ -0,0 +1,21 @@
+import { HexString } from '@polkadot/util/types';
+
+import { CONTENT_TYPE_HEADERS, METADATA_STORAGE_API_URL } from '@/shared/config';
+import { fetchWithGuard } from '@/shared/helpers';
+import { GetMetaResponse } from './types';
+
+const fetchMetadata = (hash: HexString) => {
+  const url = new URL(`${METADATA_STORAGE_API_URL}/meta`);
+  url.searchParams.append('hash', hash);
+
+  return fetchWithGuard<GetMetaResponse>(url, { method: 'GET' }).then(({ hex }) => ({ result: { hex } }));
+};
+
+const addMetadata = (hash: HexString, hex: HexString) =>
+  fetchWithGuard(`${METADATA_STORAGE_API_URL}/meta`, {
+    method: 'POST',
+    headers: CONTENT_TYPE_HEADERS,
```

Comment:
Instead of dividing common `CONTENT_TYPE_HEADERS` constant, I would incapsulate logic required for `POST` method in the `fetchWithGuard` itself. No need to have `CONTENT_TYPE_HEADERS`, as it can be hidden under the hood, as well as `JSON.stringify` for the `body`.

Should `fetchWithGuard(url: string, method: 'GET' | 'POST')` be enough for our cases? Everything else can be handled as a part of it's internal logic.